Hazel/src/Hazel/ImGui/ImGuiConsole.cpp

```diff
@@ -0,0 +1,249 @@
+			ImGui::TextUnformatted(m_Message.c_str());
+			ImGui::PopStyleColor();
+		}
+	}
+
+	ImGuiConsole::Message::Level ImGuiConsole::Message::GetLowerLevel(Level level)
+	{
+		switch (level)
+		{
+			case ImGuiConsole::Message::Level::Off     : return ImGuiConsole::Message::Level::Critical;
+			case ImGuiConsole::Message::Level::Critical: return ImGuiConsole::Message::Level::Error;
+			case ImGuiConsole::Message::Level::Error   : return ImGuiConsole::Message::Level::Warn;
+			case ImGuiConsole::Message::Level::Warn    : //return ImGuiConsole::Message::Level::Debug;
+			case ImGuiConsole::Message::Level::Debug   : return ImGuiConsole::Message::Level::Info;
+			case ImGuiConsole::Message::Level::Info    :
+			case ImGuiConsole::Message::Level::Trace   : return ImGuiConsole::Message::Level::Trace;
+		}
+		return ImGuiConsole::Message::Level::Invalid;
+	}
+
+	ImGuiConsole::Message::Level ImGuiConsole::Message::GetHigherLevel(Level level)
+	{
+		switch (level)
+		{
+			case ImGuiConsole::Message::Level::Trace   : return ImGuiConsole::Message::Level::Info;
+			case ImGuiConsole::Message::Level::Info    : //return ImGuiConsole::Message::Level::Debug;
+			case ImGuiConsole::Message::Level::Debug   : return ImGuiConsole::Message::Level::Warn;
+			case ImGuiConsole::Message::Level::Warn    : return ImGuiConsole::Message::Level::Error;
+			case ImGuiConsole::Message::Level::Error   : return ImGuiConsole::Message::Level::Critical;
+			case ImGuiConsole::Message::Level::Critical:
+			case ImGuiConsole::Message::Level::Off     : return ImGuiConsole::Message::Level::Off;
+		}
+		return ImGuiConsole::Message::Level::Invalid;
+	}
+
+
+	const char* ImGuiConsole::Message::GetLevelName(Level level)
+	{
+		switch (level)
+		{
+			case ImGuiConsole::Message::Level::Trace   : return "Trace";
+			case ImGuiConsole::Message::Level::Info    : return "Info";
+			case ImGuiConsole::Message::Level::Debug   : return "Debug";
+			case ImGuiConsole::Message::Level::Warn    : return "Warning";
+			case ImGuiConsole::Message::Level::Error   : return "Error";
+			case ImGuiConsole::Message::Level::Critical: return "Critical";
+			case ImGuiConsole::Message::Level::Off     : return "None";
+		}
+		return "Unknown name";
+	}
+
+	ImGuiConsole::Message::Color ImGuiConsole::Message::GetRenderColor(Level level)
+	{
+		switch (level)
+		{
+			case ImGuiConsole::Message::Level::Trace   : return { 0.75f, 0.75f, 0.75f, 1.00f }; // White-ish gray
+			case ImGuiConsole::Message::Level::Info    : return { 0.00f, 0.50f, 0.00f, 1.00f }; // Green
+			case ImGuiConsole::Message::Level::Debug   : return { 0.00f, 0.50f, 0.50f, 1.00f }; // Cyan
+			case ImGuiConsole::Message::Level::Warn    : return { 1.00f, 1.00f, 0.00f, 1.00f }; // Yellow
+			case ImGuiConsole::Message::Level::Error   : return { 1.00f, 0.00f, 0.00f, 1.00f }; // Red
+			case ImGuiConsole::Message::Level::Critical: return { 1.00f, 1.00f, 1.00f, 1.00f }; // White-white
```

Comment:
Also, a suggestion for the color of critical messages, maybe try magenta which is = { 1.00f, 0.00f, 1.00f, 1.00f }, just so the user can see the message more clearly.